build-config/make/crosstool-ng.make

```diff
@@ -43,24 +54,35 @@ crosstool-ng-download: $(CROSSTOOL_NG_DOWNLOAD_STAMP)
 $(CROSSTOOL_NG_DOWNLOAD_STAMP): $(PROJECT_STAMP)
 	$(Q) rm -f $@ && eval $(PROFILE_STAMP)
 	$(Q) echo "==== Getting upstream $(CROSSTOOL_NG_DESC) ===="
-	$(Q) $(SCRIPTDIR)/fetch-package $(DOWNLOADDIR) $(UPSTREAMDIR) \
-		$(CROSSTOOL_NG_TARBALL) $(CROSSTOOL_NG_URLS)
+	$(Q) if [ $(shell awk 'BEGIN{ print "'$(CROSSTOOL_NG_VERSION)'"<="'$(CROSSTOOL_NG_LATEST_RELEASE_VERSION)'" }') -eq 1 ]; then \
+			$(SCRIPTDIR)/fetch-package $(DOWNLOADDIR) $(UPSTREAMDIR) \
+			$(CROSSTOOL_NG_TARBALL) $(CROSSTOOL_NG_URLS); \
+	     fi
 	$(Q) touch $@
 
 SOURCE += $(CROSSTOOL_NG_SOURCE_STAMP)
 crosstool-ng-source: $(CROSSTOOL_NG_SOURCE_STAMP)
 $(CROSSTOOL_NG_SOURCE_STAMP): $(CROSSTOOL_NG_DOWNLOAD_STAMP)
 	$(Q) rm -f $@ && eval $(PROFILE_STAMP)
 	$(Q) echo "==== Extracting upstream $(CROSSTOOL_NG_DESC) ===="
-	$(Q) $(SCRIPTDIR)/extract-package $(CROSSTOOL_NG_BUILD_DIR) $(DOWNLOADDIR)/$(CROSSTOOL_NG_TARBALL)
+	$(Q) if [ $(shell awk 'BEGIN{ print "'$(CROSSTOOL_NG_VERSION)'"<="'$(CROSSTOOL_NG_LATEST_RELEASE_VERSION)'" }') -eq 1 ]; then \
+			$(SCRIPTDIR)/extract-package $(CROSSTOOL_NG_BUILD_DIR) \
+			$(DOWNLOADDIR)/$(CROSSTOOL_NG_TARBALL); \
+	     else \
+			git clone https://github.com/crosstool-ng/crosstool-ng \
+			$(CROSSTOOL_NG_DIR); \
+			cd $(CROSSTOOL_NG_DIR) && git reset --hard $(CROSSTOOL_NG_COMMIT);\
+	     fi	
```

Comment:
The logic here and above in the DOWNLOAD section is a bit convoluted.

I understand that in some cases we want to download and extract a release tarball and in other case we want to git clone a specific commit.  I trying to think of a way to express that concept a bit more clearly.

I do not like checking CROSSTOOL_NG_VERSION <= CROSSTOOL_NG_LATEST_RELEASE_VERSION.  That will be difficult to maintain going forward.

Let's set and check a flag whether or not to use the commit id.  If the flag is set then use git clone, if it is not set then download a released tarball.

I'm thinking for both DOWNLOAD and SOURCE targets something like this (pseudocode):

```
CROSS_TOOL_NG_USE_COMMIT  ?= no

DOWNLOAD:
  if [ "$(CROSS_TOOL_NG_USE_COMMIT)" != "yes" ] ; then
      fetch-package $(CROSSTOOL_NG_VERSION)
  fi

SOURCE:
  if [ "$CROSS_TOOL_NG_USE_COMMIT" = "yes" ] ; then
      git clone $(CROSSTOOL_NG_COMMIT)
  else
      extract-package .....
  fi

```